### ISSUE TITLE
SIGSEGV when recording a program

### ISSUE THREAD

**Issue by iacore** [rr-debugger/rr#3489]
rr crashes when recording this program.

`colors.c`
```
#include <stdio.h>

int main(void)
{
    int i, j, n;

    for (i = 0; i < 11; i++) {
        for (j = 0; j < 10; j++) {
            n = 10 * i + j;
            if (n > 108) break;
            printf("\033[%dm %3d\033[m", n, n);
        }
        printf("\n");
    }
    return 0;
}
```

```
❯ gcc -o colors colors.c
❯ rr record ./colors
rr: Saving execution to trace directory `/home/user/.local/share/rr/colors-0'.
fish: Job 1, 'rr record ./colors' terminated by signal SIGSEGV (Address boundary error)
```

It doesn't crash with root. (`rr record ./colors`)

**Comment by rocallahan** [comment]
Try again with master? I just fixed a regression that may be related.

**Comment by iacore** [comment]
It's fixed between b6dd0aa..061f041
